From c4a8e00da41c43da7e78858e06f3153f09f6967d Mon Sep 17 00:00:00 2001
From: Aliaksey Kandratsenka <alkondratenko@gmail.com>
Date: Sat, 24 Mar 2018 19:09:15 -0700
Subject: [PATCH 2/4] Fix warning about one of CheckAddressBits functions
 unused

---
 src/system-alloc.cc | 15 ++++++---------
 1 file changed, 6 insertions(+), 9 deletions(-)

diff --git a/src/system-alloc.cc b/src/system-alloc.cc
index 6b59e11..7e3f602 100755
--- a/src/system-alloc.cc
+++ b/src/system-alloc.cc
@@ -100,13 +100,11 @@ using tcmalloc::Log;
 namespace {
 
 // Check that no bit is set at position ADDRESS_BITS or higher.
-template <int ADDRESS_BITS> bool CheckAddressBits(uintptr_t ptr) {
-  return (ptr >> ADDRESS_BITS) == 0;
-}
-
-// Specialize for the bit width of a pointer to avoid undefined shift.
-template <> bool CheckAddressBits<8 * sizeof(void*)>(uintptr_t ptr) {
-  return true;
+bool CheckAddressBits(uintptr_t ptr) {
+  if (kAddressBits == 8 * sizeof(void*)) {
+    return true;
+  }
+  return (ptr >> kAddressBits) == 0;
 }
 
 }  // Anonymous namespace to avoid name conflicts on "CheckAddressBits".
@@ -510,8 +508,7 @@ void* TCMalloc_SystemAlloc(size_t size, size_t *actual_size,
   void* result = tcmalloc_sys_alloc->Alloc(size, actual_size, alignment);
   if (result != NULL) {
     CHECK_CONDITION(
-      CheckAddressBits<kAddressBits>(
-        reinterpret_cast<uintptr_t>(result) + *actual_size - 1));
+      CheckAddressBits(reinterpret_cast<uintptr_t>(result) + *actual_size - 1));
     TCMalloc_SystemTaken += *actual_size;
   }
   return result;
-- 
2.16.3

