--- src/profiler.cc.orig	2006-06-09 09:16:58.000000000 +0200
+++ src/profiler.cc	2006-08-22 22:16:53.000000000 +0200
@@ -134,6 +134,12 @@
   return (void*)sig_structure.regs->nip;
 }
 
+#elif defined HAVE_STRUCT_SIGCONTEXT_SC_PC
+typedef struct sigcontext SigStructure;
+inline void* GetPC(const SigStructure& sig_structure ) {
+  return (void*)sig_structure.sc_pc;
+}
+
 #else
 #error I dont know what your PC is
 

